src/anomalib/engine/engine.py

```diff
@@ -409,6 +410,15 @@ def _setup_anomalib_callbacks(self) -> None:
         """Set up callbacks for the trainer."""
         _callbacks: list[Callback] = [RichProgressBar(), RichModelSummary()]
 
+        # Update dirpath of each cached callback preppending "default_root_dir" 
+        # (which was updated in _setup_workspace() method)
+        for cached_callback in self._cache.args["callbacks"]:
+            if hasattr(cached_callback, "dirpath"):
+                callback_dirpath = Path(cached_callback.dirpath)
```

Comment:
is `dirpath` always the correct/fixed argument name for custom callbacks? If not, this approach may not work I'm afraid.